App/In-App Actions/URLMenuPresenter.swift

```diff
@@ -292,6 +294,72 @@ private func chromifyURL(_ url: URL) -> URL {
     return components.url!
 }
 
+func downloadVideo(with url: URL, completion: @escaping (URL?) -> ()) {
+    URLSession.shared.downloadTask(with: url) { url, response, error in
+        guard let tempUrl = url, error == nil else {
+            return completion(nil)
+        }
+  
+        let fileManager = FileManager.default
+        let documentsUrl = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first!
+        let videoFileUrl = documentsUrl.appendingPathComponent(response?.suggestedFilename ?? "temp.mp4")
```

Comment:
`downloadTask()` will delete the downloaded file once we leave this block, and I would love to avoid having to clean up after ourselves. What if we skip this moving around and just call `completion(tempUrl)`? We'd need to add to the photos library synchronously in that case, I'll add a note there too.

Reply:
On second thought, probably best not to do too much work on the URLSession delegate thread :)

Reply:
Are you sure? Happy to leave for now and sort it in the future if you like?

Reply:
Yep, I'm sure. It looks like we're cleaning up after ourselves, and keeping the shared URLSession delegate queue running is more important than us saving a line or two of code.